docs/source/pages/tutorials/detect_malicious_package.rst

```diff
@@ -0,0 +1,186 @@
+
+
+  apply_policy_to("check-django", component_id) :-
+      is_component(component_id, purl),
+      match("pkg:pypi/django@.*", purl).
+
+The ``match`` constraint in this policy allows us to apply the policy on all versions of ``django``. The result of this command should show that the policy succeeds with a zero exit code:
+
+.. code-block:: javascript
+
+    passed_policies
+        ['check-django']
+    component_satisfies_policy
+        ['1', 'pkg:pypi/django@5.0.6', 'check-django']
+    failed_policies
+    component_violates_policy
+
+Note that the ``match`` constraint applies a regex and can be expanded to enforce the ``mcn_detect_malicious_metadata_1`` check to pass on all Python packages analyzed so far by Macaron:
+
+.. code-block:: prolog
+
+  apply_policy_to("check-django", component_id) :-
+      is_component(component_id, purl),
+      match("pkg:pypi.*", purl).
+
+Note that if a policy fails to pass, Macaron returns a none-zero error code.
+
+''''''''''''''''''''''''''''''''''
+Analyzing django with dependencies
+''''''''''''''''''''''''''''''''''
+
+Macaron supports analyzing dependencies of a package and runs the same set of checks on the dependencies as the main target package. To analyze the dependencies of ``django@5.0.6`` Python package, you can either :ref:`generate an SBOM <python-sbom>` yourself or :ref:`point Macaron to a virtual environment <python-venv-deps>` where ``django`` is installed.
+
+
+Let's assume ``/tmp/.django_venv`` is the virtual environment where ``django@5.0.6`` is installed. Run Macaron as follows to analyze ``django`` and its dependencies.
+
+.. code-block:: shell
+
+  ./run_macaron.sh analyze -purl pkg:pypi/django@5.0.6 --python-venv "/tmp/.django_venv"
+
+
+By default Macaron only checks the direct dependencies. To turn on recursive dependency analysis, add the following to the ``configurations.ini`` file:
+
+.. code-block:: ini
+
+  [dependency.resolver]
+  recursive = True
+
+And pass that to the ``analyze`` command:
+
+.. code-block:: shell
+
+  ./run_macaron.sh --defaults-path configurations.ini analyze -purl pkg:pypi/django@5.0.6 --python-venv "/tmp/.django_venv"
+
+To learn more about changing configurations see :ref:`here <change-config>`.
+
+Now we can enforce the policy below to ensure that the ``mcn_detect_malicious_metadata_1`` check always passes on ``django`` and its dependencies and none of the dependencies have malicious behavior.
+
+.. code-block:: prolog
+
+  #include "prelude.dl"
+
+  Policy("check-dependencies", component_id, "Check the dependencies of django.") :-
+    is_component(component_id, _),
+    transitive_dependency(component_id, dependency),
+    check_passed(dependency, "mcn_detect_malicious_metadata_1").
+
+  apply_policy_to("check-dependencies", component_id) :-
+      is_component(component_id, purl),
+      match("pkg:pypi.*", purl).
+
+As you can see below, the policy passes for ``django`` and all its transitive dependencies.
```

Comment:
```suggestion
As you can see below, the policy passes for ``django`` because it doesn't detect malicious behavior for `django` and all its transitive dependencies.
```
Because the target of the Policy is `django`, using `and all its transitive dependencies.` create an impression for me that each transitive dependency is also a target.

Reply:
I agree, except it should end with `or any of its transitive dependencies.`

Reply:
See commit [728d753](https://github.com/oracle/macaron/pull/816/commits/728d753678ee948b303d9b2fb9c672f245ed7ee0)